### PR TITLE
SDCB-9997 [Appium 2.0] Java Server Side sample

### PR DESCRIPTION
**Pull request by eryk-tkaczewski**

### CODE REVIEW

**Review by michal-przytarski — CHANGES_REQUESTED** [review]
In `README.md` in "client side" there are capabilities described with `_` instead of `:`

Also caps are wrongly managed in there 

For iOS:

```
{"capabilities":{"firstMatch":[{}],"alwaysMatch":{"appium:app":"/Users/testdroid/test/application.ipa","appium:options":{"deviceName":"iOS Phone"},"bitbar:options":{"automationName":"XCUITest"},"platformName":"IOS"}}}
```

Android:

```
{"capabilities":{"firstMatch":[{}],"alwaysMatch":{"appium:app":"/test/application.apk","appium:options":{"deviceName":"Android Device"},"bitbar:options":{},"platformName":"ANDROID"}}}
```

Automation name is not BitBar option. BitBar options shouldn't be present in CS tests

**Review by michal-przytarski — APPROVED** [review]
Approving just to let somebody else merge this while I'm out of office. Please fix it anyway.

File: samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ private DesiredCapabilities getDesiredCapabilitiesFromProperties() throws IOExce
         LOGGER.debug("Setting client side specific capabilities... FINISHED");
     } else if (isServerSideTestRun()) {
         LOGGER.debug("Setting server side specific capabilities...");
-        bitbarOptions.setCapability("app", getServerSideApplicationPath());
+        desiredCapabilities .setCapability("app", getServerSideApplicationPath());
```
[the remark this answers is not in the document]
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/client-side/java/src/test/java/com/testdroid/appium/BaseTest.java]: done

**Comment by krzysztof-moscichowski-smartbear** [comment]
Please squash commits.

File: samples/testing-frameworks/appium/server-side/java/src/test/java/AbstractAppiumTest.java [hunk 2]
```diff
@@ -56,32 +56,16 @@ public static AppiumDriver getIOSDriver() throws Exception {
...
-            capabilities.setCapability("udid", udid);
+            appiumOptions.setCapability("udid", udid);
         }
-        if (platformVersion != null) {
-            capabilities.setCapability("platformVersion", platformVersion);
-        }
-        capabilities.setCapability("app", appFile);
-        capabilities.setCapability("newCommandTimeout", 120);
+        appiumOptions.setCapability("app", appFile);
+        capabilities.setCapability("appium:options", appiumOptions);
```
Comment (krzysztof-moscichowski-smartbear) [inline samples/testing-frameworks/appium/server-side/java/src/test/java/AbstractAppiumTest.java:68]: Please sort order of lines so appiumOptions and capabilities go together.

File: samples/testing-frameworks/appium/server-side/java/src/test/java/AbstractAppiumTest.java [hunk 3]
```diff
@@ -91,23 +75,14 @@ public static AppiumDriver getIOSDriver() throws Exception {
...
-        }
-        if (platformVersion != null) {
-            capabilities.setCapability("platformVersion", platformVersion);
-        }
-
-        capabilities.setCapability("app", appFile);
-        capabilities.setCapability("newCommandTimeout", 120);
+        appiumOptions.setCapability("deviceName", "Android Device");
+        appiumOptions.setCapability("app", appFile);
+        capabilities.setCapability("appium:options", appiumOptions);
```
Comment (krzysztof-moscichowski-smartbear) [inline samples/testing-frameworks/appium/server-side/java/src/test/java/AbstractAppiumTest.java:86]: Please sort order of lines so appiumOptions and capabilities go together.

File: samples/testing-frameworks/appium/server-side/java/src/test/java/AbstractAppiumTest.java (hunk 2 above)
Comment (eryk-tkaczewski) [reply samples/testing-frameworks/appium/server-side/java/src/test/java/AbstractAppiumTest.java:68]: Done

**Review by krzysztof-moscichowski-smartbear — APPROVED** [review]

**Pull request merged**